### PR TITLE
Refactor Relationships and Descriptions in Work Show. #1272

### LINKED ISSUE #1272: Refactor Relationships and Descriptions in Work Show
### Descriptive summary

In Hyrax 2.0.0 alpha the relationships and descriptions for the work show are in tables. Our screen reader user did not expect them to be in tables and instead suggested `<dd><dl>` and `<ul><li>`.

### Rationale

Tables were unexpected and created a bit of confusion after creating a work and getting to the show page. She expected the metadata to be in lists.

### Expected behavior

Metadata about a work to be in lists.

### Actual behavior

Metadata and relationships are in a table.

### Steps to reproduce the behavior

1. View a work
1. Inspect source for Metadata and Relationships

### Related work

https://github.com/psu-stewardship/scholarsphere/pull/680
https://github.com/psu-stewardship/scholarsphere/pull/724/files

### PR DESCRIPTION
**Pull request by hweng**
Fixes #1272 ; refs #1272

Refactor Relationships and Descriptions in Work Show.

@samvera/hyrax-code-reviewers

### CODE REVIEW

**Comment by cjcolvar** [comment]
@adamjarling @mtribone Would you two be able to review this?
@hweng Could you please post screenshots of before/after?

File: app/renderers/hyrax/renderers/attribute_renderer.rb [hunk 1]
```diff
@@ -33,6 +33,20 @@ def render
         markup.html_safe
       end
 
+      # Draw the dl row for the attribute
+      def render_dl_row
+        markup = ''
+
+        return markup if values.blank? && !options[:include_empty]
+        markup << %(<dt>#{label}</dt>\n<dd><ul class='tabular'>)
```
Comment (mtribone) [inline app/renderers/hyrax/renderers/attribute_renderer.rb]: I don't believe that an unordered list is permissible inside of the `<dd>` and am unsure of the context of its use. Can you use multiple `<dd>` instead? For example, single term and multiple descriptions: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dl#Examples

File: app/views/hyrax/base/_relationships_parent_row.html.erb [hunk 2]
```diff
@@ -1,6 +1,5 @@
-<tr>
-  <th><%= t(".label", type: type.humanize) %></th>
-  <td>
+<dt> <%= t(".label", type: type.humanize) %> </dt>
+<dd>
```
Comment (mtribone) [inline app/views/hyrax/base/_relationships_parent_row.html.erb]: Similar comment here about the unordered list inside of a description list. The relationships aren't key/value pairs and could be just an unordered list. It's a little different than metadata.

File: app/views/hyrax/base/_relationships.html.erb [hunk 3]
```diff
@@ -1,6 +1,4 @@
 <h2><%= t('.header') %></h2>
-<table class="table table-borderless relationships">
-  <tbody>
-    <%= render 'relationships_parent_rows', presenter: presenter %>
-  </tbody>
-</table>
+<dl class="work-show">
```
Comment (mtribone) [inline app/views/hyrax/base/_relationships.html.erb]: This could just be an unordered list. See below.

**Comment by hweng** [comment]
@adamjarling @mtribone @cjcolvar Thanks for reviewing it.  Please see the screenshots of before/after.

Before the change:
before-change

After the change:
after-change

**Comment by hweng** [comment]
The attribute_renderer.rb and presents_attributes.rb are updated to provide the option to draw the dl row in addition to HTML table row, so that in view page it can add property `html_dl: true` if one decide to use <dl> instead of <table>

**Comment by hweng** [comment]
@mtribone Here is the HTML result for RELATIONSHIPS. It looks fine to me that the unordered list is inside of the <dd>. But it’s also fine for me to update it to use multiple <dt><dd> instead. Let me know what all of you think about it?

Before changes:
```
<h2>Relationships</h2>
<table class="table table-borderless relationships">
<tbody>
<tr>
<th>In Collection:</th>
<td>
<ul class="tabular">
<li class='attribute attribute-title'>
<a href="/collections/4t64gn166?locale=en">animal</a>
</li>
</ul>
</td>
</tr>
<tr>
<th>In Administrative Set:</th>
<td><ul class='tabular'><li class="attribute attribute-admin_set"><a href="/catalog?f%5Badmin_set_sim%5D%5B%5D=Default+Admin+Set">Default Admin Set</a>
</li></ul></td></tr>
</tbody>
</table>
```

After Changes:

```
<h2>Relationships</h2>
<dl class="work-show">
<dt> In Collection: </dt>
<dd>
<ul class="tabular">
<li class='attribute attribute-title'>
<a href="/collections/4t64gn166?locale=en">animal</a>
</li>
</ul>
</dd>
<dt>In Administrative Set:</dt>
<dd>
<ul class='tabular'>
<li class="attribute attribute-admin_set"><a href="/catalog?f%5Badmin_set_sim%5D%5B%5D=Default+Admin+Set">Default Admin Set</a><
/li>
</ul>
</dd>
</dl>

```

**Comment by mtribone** [comment]
@hweng Looks good. Initially I didn't think that nesting the `<ul>` inside of the `<dd>` would be valid HTML, but looking again at the spec it does allow "flow content" for the `<dd>` element.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dd

**Comment by cjcolvar** [comment]
@julesies @vantuyls does this change look okay to you?  See [screenshots above](https://github.com/samvera/hyrax/pull/2784#issuecomment-376680009).

**Comment by julesies** [comment]
This seems to change the way MD values are displayed? Hopefully @vantuyls can weigh in on that since it is a visual UI change.

**Comment by vantuyls** [comment]
I'm fine with the change.

**Comment by hweng** [comment]
@adamjarling @mtribone @julesies @vantuyls 
Need one of your approve again.  Not sure why it dismissed the approving reviews from @mtribone and @adamjarling. I was just using the web editor to resolve the conflicts and merged master branch into this branch. Thank you all for your reviews.

**Comment by adamjarling** [comment]
Hi @hweng I'll wait for the build to complete, then give it another look at get it approved for you.

**Comment by adamjarling** [comment]
@hweng If there'd be a way to squash all 4 commits into one commit, that'd be great.  @cjcolvar thoughts?

**Comment by hweng** [comment]
@adamjarling Yes, there is "Squash and merge" option in the drop down menu if you click the Down Arrow button next to "Merge pull request". Let me know if you would like me to do it or you will merge it?

**Comment by adamjarling** [comment]
@hweng Sounds good.  Go for it.

**Pull request merged**